pkg/arvo/sys/hoon.hoon

```diff
@@ -226,6 +226,33 @@
     *   (add (mod b 2) (mul $(b (div b 2)) 2))
   ==
 ::
+++  pin
+  ~/  %pin
+  ::    sub axis test
+  ::
+  ::  checks whether {b} addresses a subtree of the tree addressed by {a}.
+  |=  [a=@ b=@]
+  ?<  =(0 a)
+  ?<  =(0 b)
+  |-  ^-  ?
+  ?:  =(a 1)  %&
+  ?:  =((cap a) (cap b))
+    $(a (mas a), b (mas b))
+  %|
```

Comment:
To ensure that the last sub-hoon is the largest, can you switch this to a `?.`? Also, this needs a single `::` line between the `+pin` and `+awl` arms.